devlib/target.py

```diff
@@ -701,6 +703,33 @@ def _install_module(self, mod, **params):
     def _resolve_paths(self):
         raise NotImplementedError()
 
+    def is_network_connected(self):
+        self.logger.debug('Checking for internet connectivity...')
+        return self._ping_server(GOOGLE_DNS_SERVER_ADDRESS)
+
+    def _ping_server(self, ip_address, timeout=1, packet_count=1):
```

Comment:
why not exposing this also as a public API?

Reply:
Should not be safer to usage a slightly longer timeout... maybe 3s?

Reply:
> why not exposing this also as a public API?

I think keeping this internal for now is the right move. We can always open it up later, when there is a clear need.